### PR TITLE
[Entry] Dart Map Concept

### PR DESCRIPTION
**Pull request by t-p-gregory**
### Description

- Added syntax and example for Map concept in Dart language.
- Closes issue #4244 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/dart/concepts/map/map.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+Title: 'Map'
+Description: 'The Map object is a simple key/value pair. Keys and values in a map may be of any type. A Map is a dynamic collection. In other words, Maps can grow and shrink at runtime.'
```
Comment (dakshdeepHERE) [inline content/dart/concepts/map/map.md:3]: Give a small and breif desc here. This big definition about map can go below this table(meta data).

File: content/dart/concepts/map/map.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+Subjects:
+  - 'Map term in Dart'
```
Comment (dakshdeepHERE) [inline content/dart/concepts/map/map.md:5]: Only use the subjects present in this [file](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md)

File: content/dart/concepts/map/map.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  - 'Map'
+  - 'Dart'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+## Syntax
+
+The syntax to declare a map is -
```
Comment (dakshdeepHERE) [inline content/dart/concepts/map/map.md:16]:
```suggestion
The syntax to declare a `map` is:
```

File: content/dart/concepts/map/map.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+'''
```
Comment (dakshdeepHERE) [inline content/dart/concepts/map/map.md:18]: The tag should start with ``` and make sure to give the pseudo tag for syntax and dart tag for dart code snippets below. Here the tag should be `pseudo`.

File: content/dart/concepts/map/map.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+Here is an example -
+
+'''dart
+void main() { 
+   var details = {'Cat':'jeff','Color':'brown'}; 
+   print(details); 
+}
+'''
+
+...producing the following output -
```
Comment (dakshdeepHERE) [inline content/dart/concepts/map/map.md:33]:
```suggestion
The above code gives the following output:
```

File: content/dart/concepts/map/map.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+
+'''
```
Comment (dakshdeepHERE) [inline content/dart/concepts/map/map.md:36]: Use shell tag here. The brackets should be ``` not '''.

File: content/dart/concepts/map/map.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+The syntax to declare a map is -
+
+'''
+var identifier = new Map()
+'''
+
+## Example
+
+Here is an example -
```
Comment (dakshdeepHERE) [inline content/dart/concepts/map/map.md:24]:
```suggestion
Here is an example on how to use `map` in Dart:
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
For now work on these comments @t-p-gregory. Overall good entry 🚀

File: content/dart/concepts/map/map.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'Map'
+  - 'Dart'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
```
Comment (dakshdeepHERE) [inline content/dart/concepts/map/map.md:13]: give a description here on what map are.

File: content/dart/concepts/map/map.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+## Syntax
+
+The syntax to declare a `map` is -
```
Comment (dakshdeepHERE) [inline content/dart/concepts/map/map.md:16]:
```suggestion
The syntax to declare a `map` is:
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
2 more changes

File: content/dart/concepts/map/map.md (hunk 9 above)
Comment (t-p-gregory) [reply content/dart/concepts/map/map.md]: The changes have been made. Please let me know anything else I can improve. Thank you!

**Review by dakshdeepHERE — APPROVED** [review]
LGTM 🚀

File: content/dart/concepts/map/map.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  - 'Code Foundations'
+Tags:
+  - 'Map'
+  - 'Dart'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+The Map object is a key/value pair. The values and keys in a map may be of any type. A Map is a dynamic collection that can grow and shrink at runtime.
```
Comment (ishg-153) [inline content/dart/concepts/map/map.md:15]:
```suggestion
The **Map** object is a key/value pair. The values and keys in a map may be of any type. A Map is a dynamic collection that can grow and shrink at runtime.
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@t-p-gregory One small change from my end.

**Comment by ishg-153** [comment]
Going ahead and making the change myself to speed up the process

**Review by ishg-153 — APPROVED** [review]
@t-p-gregory this is good to merge.

**Pull request merged**